api/ruby/build-your-first-github-app/advanced_server.rb

```diff
@@ -0,0 +1,158 @@
+require 'sinatra'
+require 'logger'
+require 'json'
+require 'openssl'
+require 'octokit'
+require 'jwt'
+require 'time' # This is necessary to get the ISO 8601 representation of a Time object
+
+#
+#
+# This is a customized server for the GitHub App you can build by following
+# https://developer.github.com/build-your-first-github-app.
+#
+#
+
+class GHAapp < Sinatra::Application
+
+# Never, ever, hardcode app tokens or other secrets in your code!
+# Always extract from a runtime source, like an environment variable.
+
+
+# Notice that the private key must be in PEM format, but the newlines should be stripped and replaced with
+# the literal `\n`. This can be done in the terminal as such:
+# export GITHUB_PRIVATE_KEY=`awk '{printf "%s\\n", $0}' private-key.pem`
+  PRIVATE_KEY = OpenSSL::PKey::RSA.new(ENV['GITHUB_PRIVATE_KEY'].gsub('\n', "\n")) # convert newlines
+
+# You set the webhook secret when you create your app. This verifies that the webhook is really coming from GH.
+  WEBHOOK_SECRET = ENV['GITHUB_WEBHOOK_SECRET']
```

Comment:
A commented .env.example file might be helpful as an alternative to exporting 
- GITHUB_PRIVATE_KEY 
- GITHUB_WEBHOOK_SECRET
- GITHUB_APP_IDENTIFIER